hollow/src/main/java/com/netflix/hollow/api/objects/provider/HollowObjectCacheProvider.java

```diff
@@ -84,11 +84,18 @@ public HollowObjectCacheProvider(HollowTypeDataAccess typeDataAccess, HollowType
 
     @Override
     public T getHollowObject(int ordinal) {
-        return cachedItems.get(ordinal);
+        List<T> refCachedItems = cachedItems;
+        if (refCachedItems == null) {
+            throw new IllegalStateException(String.format("HollowObjectCacheProvider for type %s has been detached or was not initialized", typeReadState == null ? null : typeReadState.getSchema().getName()));
+        }
+        if (refCachedItems.size() <= ordinal) {
+            throw new IllegalStateException(String.format("Ordinal %s is out of bounds for pojo cache array of size %s.", ordinal, refCachedItems.size()));
+        }
+        return refCachedItems.get(ordinal);
```

Comment:
Is the return type for getHollowObject nullable here? Is that expected? When does it become null?

Reply:
yes if for e.g.`api.getMovie(ordinal)` is called with an ordinal that isn't populated that will return a null- that seems reasonable. This behavior from `HollowObjectCacheProvider` (for cached types) is different from behavior in `HollowObjectFactoryProvider` (non cached types) where `HollowObjectFactoryProvider` will return garbage results if called with ordinals that aren't populated - i dont think thats ideal but it likely doesn't matter- its basically saying responsibility is on user of low level api (when passing ordinal) to ensure that it is called with ordinal that is populated current state.